src/avram/database.cr

```diff
@@ -3,7 +3,8 @@ abstract class Avram::Database
 
   @@db : DB::Database? = nil
   @@lock = Mutex.new
-  class_getter transactions = {} of FiberId => DB::Transaction
+  class_getter connections = {} of FiberId => DB::Connection
```

Comment:
We maintain the connections reference instead of transactions because new sub-transactions can be made which would have overwrote the parent transaction. By holding onto the connection (and the added monkeypatch) we can  still get to the transaction

Reply:
How is this hash cleared? I think it might prevent Connection objects from being released. WeakRef can help in that regard if there is no good place to clear the hash.

Reply:
I don't know if the connections hash was ever being cleared. I think the individual connections inside were just being deleted somewhere... 🤔  I'd like to learn more about this `WeakRef`. I'm still not really sure what it does, or how to use it